pppd/pppd.h

```diff
@@ -235,6 +235,8 @@ struct channel {
 	void (*send_config)(int, uint32_t, int, int);
 	/* set the receive-side PPP parameters of the channel */
 	void (*recv_config)(int, uint32_t, int, int);
+	/* close discovery sock */
+	void (*close_discovery_sock)(void);
```

Comment:
Why go to the length of adding another callback? It could seem like registering a "cleanup" function would be suficient.